crates/bevy_asset/src/ser.rs

```diff
@@ -0,0 +1,107 @@
+use std::cell::Cell;
+
+use bevy_reflect::Uuid;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::{Asset, AssetServer, Handle, HandleId, HandleUntyped};
+
+///////////////////////////////////////////////////////////////////////////////
+
+thread_local! {
+    static ASSET_SERVER: Cell<Option<AssetServer>> = Cell::new(None);
+}
+
+#[derive(Serialize, Deserialize)]
+enum AssetRef {
+    Default,
+    /// Used for static handles like the `PBR_PIPELINE_HANDLE` or a embedded assets
+    Local(Uuid, u64),
+    /// Loads from a file
+    External(String),
+}
+
+impl<T: Asset> Serialize for Handle<T> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let path = ASSET_SERVER
+            .with(|cell| {
+                let server = cell.replace(None);
+                let path = server.as_ref().and_then(|server| {
```

Comment:
If `Handle<T>` (de)serialization is not done through the `AssetServer` we should probably just fail out and `panic!`, not silently succeed.

Just call 
```rust
let server = cell.replace(None).expect("Handle (de)serialization is only supported via the `AssetServer`);
```